### PR TITLE
Feature:Social Login

### PR DESCRIPTION
**Pull request by eazy4sng**
- Add Email and Password Auth
- Add Github OAuth
- Add Google OAuth

### CODE REVIEW

File: file3.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+hello
```
Comment (pseudoeazy) [inline file3.txt:1]: This line is good but it is better if you capitalise H

**Review by pseudoeazy — CHANGES_REQUESTED** [review]
Please make these changes and do another push

File: file3.txt (hunk 1 above)
Comment (eazy4sng) [reply file3.txt]: Aside the Capital H, is there any additional stuff?

**Review by pseudoeazy — APPROVED** [review]
Everything looks good

**Pull request merged**